app/src/androidTest/java/com/termux/app/TermuxTest.java

```diff
@@ -0,0 +1,128 @@
+package com.termux.app;
+
+import android.util.Log;
+import android.view.View;
+
+import androidx.test.espresso.PerformException;
+import androidx.test.espresso.UiController;
+import androidx.test.espresso.ViewAction;
+import androidx.test.espresso.util.HumanReadables;
+import androidx.test.espresso.util.TreeIterables;
+import androidx.test.ext.junit.runners.AndroidJUnit4;
+import androidx.test.filters.LargeTest;
+import androidx.test.rule.ActivityTestRule;
+
+import com.termux.R;
+import com.termux.view.TerminalView;
+
+import org.hamcrest.Matcher;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import static androidx.test.espresso.Espresso.onView;
+import static androidx.test.espresso.action.ViewActions.closeSoftKeyboard;
+import static androidx.test.espresso.action.ViewActions.typeText;
+import static androidx.test.espresso.matcher.ViewMatchers.isAssignableFrom;
+import static androidx.test.espresso.matcher.ViewMatchers.isRoot;
+import static androidx.test.espresso.matcher.ViewMatchers.withId;
+
+
+@RunWith(AndroidJUnit4.class)
+@LargeTest
+public class TermuxTest {
+
+    private String mStringToBetyped;
+
+    @Rule
+    public ActivityTestRule<TermuxActivity> mActivityRule = new ActivityTestRule<>(
+        TermuxActivity.class);
+
+    @Before
+    public void initValidString() {
+        // Specify a valid string.
+        mStringToBetyped = "ls -lah\n";
+    }
+
+    @Test
+    public void changeText_sameActivity() {
+        // wait during 15 seconds for a view
+        onView(isRoot()).perform(waitId(R.id.terminal_view, TimeUnit.SECONDS.toMillis(15)));
```

Comment:
try a much longer wait time, the view might be taking a long time to populate on CI

```suggestion
        onView(isRoot()).perform(waitId(R.id.terminal_view, TimeUnit.SECONDS.toMillis(300)));
```

Reply:
I tried it, but as is looks like Espresso has a build-in-function to not wait too long

> Error performing 'wait for a specific view with id <2131034142> during 300000 millis.' on view 'is a root view.'.

Independent of this, when a view doesn't appear within 15/30 seconds, something other is wrong. 
Why it works with `Run` in IDE, and not with [command line](https://github.com/termux/termux-app/pull/1566#issuecomment-629787966) ?